src/DotNetty.Buffers/ReadOnlyByteBufferStream.cs

```diff
@@ -23,24 +23,22 @@ public ReadOnlyByteBufferStream(IByteBuffer buffer, bool releaseReferenceOnClosu
 
         public override bool CanRead => true;
 
-        public override bool CanSeek => false;
+        public override bool CanSeek => true;
 
         public override bool CanWrite => false;
 
-        public override long Length
-        {
-            get { throw new NotSupportedException(); }
-        }
+        public override long Length => this.buffer.WriterIndex;
 
         public override long Position
         {
-            get { throw new NotSupportedException(); }
-            set { throw new NotSupportedException(); }
+            get => this.buffer.ReaderIndex;
+            set => this.buffer.SetReaderIndex((int)value);
         }
 
         public override int Read(byte[] output, int offset, int count)
         {
-            int read = Math.Min(count - offset, this.buffer.ReadableBytes);
+            if (offset + count > output.Length) throw new ArgumentException();
```

Comment:
per coding style pls show code block explicitly (curly braces). Also, pls add a msg like `$"The sum of {nameof(offset)} and {nameof(count)} is larger than the {nameof(output)} length"`.